src/SelfAssessment/AssessmentQuestion.js

```diff
@@ -129,14 +114,26 @@ export const AssessmentQuestion = ({ onNext, onChange, option, question }) => {
           <GlobalText style={style.instruction}>
             {assessmentInputInstruction}
           </GlobalText>
-          <View style={style.optionsWrapper}>{options}</View>
+          <View style={style.optionsWrapper}>
+            <AssessmentOptions
+              option={option}
+              question={question}
+              selectedValues={selectedValues}
+              onSelect={onSelectHandler}
+            />
+          </View>
         </View>
       </ScrollView>
       <View style={style.footer}>
         <Button
-          disabled={!selectedValues.length}
-          onPress={onNext}
+          onPress={handleOnNextPress}
           label={t("assessment.next")}
+          disabled={
+            !(
+              question.question_type === QUESTION_TYPE_MULTI ||
+              selectedValues.length
```

Comment:
this `!selectedValues.length` seems to be repeated a couple of times across, nothing wrong with it, I was just wondering if there might be a concept behind it that could be better expressed as a variable?

Reply:
Good point, but since we repeat this only two times in the whole component lets just leave it as is for now.